R/eval-walk.R

```diff
@@ -286,12 +286,23 @@ call_kind <- function(expr, error_call) {
   fn <- as_string(head)
 
   if (fn %in% c("$", "[[") && identical(node_cadr(expr), quote(.data))) {
-    lifecycle::deprecate_warn("1.2.0",
-      what = I("Use of .data in tidyselect expressions"),
-      with = I("`any_of()` or `all_of()`")
-    )
-
     validate_dot_data(expr, error_call)
+
+    what <- I("Use of .data in tidyselect expressions")
+    if (fn == "$") {
+      var <- as.character(expr[[3]])
```

Comment:
I'd use:

```suggestion
      var <- as_string(expr[[3]])
```

Or, to be extremely defensive in case of metaprogramming gone wrong, `as_label()`.

Reply:
I think `as_string()` is fine since `validate_dot_data()` checks that it's not something weird.